website_product_attribute_filter/controllers/main.py

```diff
@@ -67,11 +64,27 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
                     attrib_category_ids:
                 attrib_category_ids.append(
                     attributes_ids_all[i].category_id.id)
-            for v in attributes_ids_all[i].value_ids:
-                actual_domain = domain + \
-                    [('attribute_line_ids.value_ids', 'in', [v.id])]
-                variant_count.update(
-                    {v.id: Product.search_count(actual_domain)})
+
+        value_ids = request.env['product.attribute.value'].search(
+            [('attribute_id', 'in', attributes_ids_all.mapped('id'))])
+        if value_ids:
+            attributes_ids = value_ids.mapped('id')
+            registry = odoo.modules.registry.Registry(request.session.db)
+            new_cr = registry.cursor()
+            new_cr.execute("""select product_attribute_value_id,count(
+            distinct product_template.id) as pr_count \
+            from product_product \
+            inner join product_attribute_value_product_product_rel \
+            on product_attribute_value_product_product_rel.product_product_id=
+                product_product.id
+            inner join product_template on product_template.id=
+                product_product.product_tmpl_id\
+            where sale_ok is True and is_published is True and
+                product_product.active is True and
+                product_attribute_value_id in %s \
+            group by product_attribute_value_id""", [tuple(attributes_ids)])
```

Comment:
This is completely ignoring the search domain / terms.
We don't want that.

Performance should be improved, but behaviour should be the same.

I get the loop could be the cause of the performance issue, but why don't you try to search the products with the search_domain, first, to get their ids, and then use them on the raw SQL? That way we wouldn't be ignoring `_get_search_domain` result

Reply:
Okay @ivantodorovich 

Noted it, I will forward this to Nirav.